### PR TITLE
Ec2 deployment

### PR DESCRIPTION
**Pull request by antara678**
*Issue #, if available:*

*Description of changes:*
-Added code to deploy model on ec2
-modified debug.sh to uninstall fmbench and rerun in local mode 
-added config file to run llama3 on ec2 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/fmbench/scripts/ec2_llama3_deploy.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+# globals
+# HF_TOKEN_FNAME: str = os.path.join(os.path.dirname(os.path.realpath(__file__)), "hf_token.txt")
+def set_up(model_name, model_id):    
+    directory = os.path.join(os.path.expanduser("~/foundation-model-benchmarking-tool"), model_name)
+    os.makedirs(directory, exist_ok=True)
+    # Write the content to the serving.properties file
+    content =  """\
+engine=MPI
```
Comment (aarora79) [inline src/fmbench/scripts/ec2_llama3_deploy.py:20]: The `serving.properties` content needs to be read from the config file, please see how this is handled in the other deployment scripts.

File: src/fmbench/scripts/ec2_llama3_deploy.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+option.tensor_parallel_degree=1
+option.max_rolling_batch_size=256
+option.model_id={model_id}
+option.rolling_batch=lmi-dist"""
+    file_path = os.path.join(directory, "serving.properties")
+    with open(file_path, "w") as file:
```
Comment (aarora79) [inline src/fmbench/scripts/ec2_llama3_deploy.py:26]: use `Path(fpath).write_text()`

File: debug.sh [hunk 3]
```diff
@@ -16,5 +17,5 @@ pip install -U dist/*.whl
 
 # run the newly installed version
 echo "going to run fmbench now"
-fmbench --config-file $CONFIG_FILE_PATH  > $LOGFILE 2>&1
+fmbench --config-file $CONFIG_FILE_PATH --local-mode yes --write-bucket abstaticwebsitetest1 > $LOGFILE 2>&1
```
[the remark this answers is not in the document]
Comment (antara678) [reply debug.sh]: done, reverted to old debug.sh

File: src/fmbench/scripts/deploy_model.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# script for deploying model in EC2
+# Download the model from the container and deploy
+echo "going to download model now"
+echo "content nin docket command: $REGION, $IMAGE_URI, $MODEL_NAME, $HF_TOKEN"
```
[the remark this answers is not in the document]
Comment (antara678) [reply src/fmbench/scripts/deploy_model.sh]: deleted file

File: src/fmbench/scripts/ec2_llama3_deploy.py
[the remark this answers is not in the document]
Comment (antara678) [reply src/fmbench/scripts/ec2_llama3_deploy.py:1]: renamed

File: src/fmbench/scripts/ec2_llama3_deploy.py (hunk 1 above)
Comment (antara678) [reply src/fmbench/scripts/ec2_llama3_deploy.py:20]: updated to read from config file

**Review by aarora79 — APPROVED** [review]

**Pull request merged**